app/grandchallenge/evaluation/models.py

```diff
@@ -640,7 +637,34 @@ def submission_period_is_open_now(self):
 
     @property
     def open_for_submissions(self):
-        return self.submission_period_is_open_now and self.submission_limit > 0
+        return (
+            self.submission_period_is_open_now
+            and self.submission_limit > 0
+            and not self.submission_limit_reached
+        )
+
+    @property
+    def submission_limit_reached(self):
+        return (
+            self.percent_of_submission_limit
+            and self.percent_of_submission_limit == 100
+        )
+
+    @property
+    def percent_of_submission_limit(self):
+        if self.number_of_submissions_limit:
+            return min(
+                round(
+                    (
+                        self.submission_set.count()
+                        / self.number_of_submissions_limit
+                    )
+                    * 100
+                ),
+                100,
+            )
```

Comment:
Fair enough, might indeed be useful for currently ongoing challenge phases (once we've set a limit for them that is). I had set the limit to 100 because of the progress bar I use to display these, but that's not a strict requirement of course. 

If I understand the last sentence here correctly, it should not be necessary to turn this into a cached_property since we're prefetching the `submission_set` for all challenge phases in all views where this is used https://docs.djangoproject.com/en/4.1/ref/models/querysets/#django.db.models.query.QuerySet.count  Did I understand this correctly?

Reply:
You can double check by either using the debug toolbar or by using `python manage.py shell_plus --print-sql`

Reply:
We're good! No unnecessary duplicate queries.